lua/harpoon/cmd-ui.lua

```diff
@@ -0,0 +1,148 @@
+local harpoon = require("harpoon")
+local popup = require("popup")
+local log = require("harpoon.dev").log
+local term = require("harpoon.term")
+
+local M = {}
+
+Harpoon_cmd_win_id = nil
+Harpoon_cmd_bufh = nil
+
+local function close_menu(force_save)
+    force_save = force_save or false
+    local global_config = harpoon.get_global_settings()
+
+    if global_config.save_on_toggle or force_save then
+        require("harpoon.cmd-ui").on_menu_save()
+    end
+
+    vim.api.nvim_win_close(Harpoon_cmd_win_id, true)
+
+    Harpoon_cmd_win_id = nil
+    Harpoon_cmd_bufh = nil
+end
+
+local function create_window()
+    log.trace("_create_window()")
+    local config = harpoon.get_menu_config()
+    local width = config.width or 60
+    local height = config.height or 10
+    local borderchars = config.borderchars
+        or { "─", "│", "─", "│", "╭", "╮", "╯", "╰" }
+    local bufnr = vim.api.nvim_create_buf(false, false)
+
+    local Harpoon_cmd_win_id, win = popup.create(bufnr, {
+        title = "Harpoon Commands",
+        highlight = "HarpoonWindow",
+        line = math.floor(((vim.o.lines - height) / 2) - 1),
+        col = math.floor((vim.o.columns - width) / 2),
+        minwidth = width,
+        minheight = height,
+        borderchars = borderchars,
+    })
+
+    vim.api.nvim_win_set_option(
+        win.border.win_id,
+        "winhl",
+        "Normal:HarpoonBorder"
+    )
+
+    return {
+        bufnr = bufnr,
+        win_id = Harpoon_cmd_win_id,
+    }
+end
+
+local function is_white_space(str)
+    local white_space_chars = {
```

Comment:
we could probably just use \s and a regex here.  I KNOW... I SAID THAT... 

There should be a gsub call you can do, something like:

```
return #(str:gsub('%s', '')) == 0 -- not sure the exact api
```

Reply:
done the change